src/api.rs

```diff
@@ -1369,6 +1495,67 @@ impl Client {
     }
 }
 
+fn start_sso_callback_server(
+    listener: &TcpListener,
+    state: &str,
+) -> Result<String> {
+    let (mut stream, _) = listener.accept().unwrap();
+
+    let buf_reader = BufReader::new(&mut stream);
+    let http_status_line: String =
+        buf_reader.lines().next().unwrap().unwrap();
+
+    let absolute_url = "http://localhost".to_string()
+        + http_status_line.split_whitespace().nth(1).unwrap();
+    let url = Url::try_from(absolute_url.as_str()).unwrap();
+
+    let mut query_pairs = url.query_pairs().into_owned();
+    let sso_code = query_pairs
+        .find(|pair| pair.0 == "code")
+        .ok_or(Error::FailedToProcessSSOCallback {
+            msg: "Could not obtain code from the URL".to_string(),
+        })?
+        .1;
+    let received_state = query_pairs
+        .find(|pair| pair.0 == "state")
+        .ok_or(Error::FailedToProcessSSOCallback {
+            msg: "Could not obtain state from the URL".to_string(),
+        })?
+        .1;
+
+    let states_match = received_state.split("_identifier=").next().unwrap()
+        == state.split("_identifier=").next().unwrap();
```

Comment:
why are we splitting `state` here? isn't it just a purely random value?

Reply:
Hmm, tbh I ported it as-is from the official implementation, it might be that it's used in more contexts there?

As indeed, this split doesn't make sense, `state` is purely random and is always alphanumeric, so we won't get any collisions with `_identitifer=`. I'll remove it